packages/plugins/src/SharedConfigPlugin.ts

```diff
@@ -43,17 +43,31 @@ export interface SharedConfigPluginOptions {
  * It then exports a JSON file (name: `options.filename`) into each directory with the merged config for that level
  */
 const SharedConfigPlugin: PluginType<SharedConfigPluginPage, SharedConfigPluginOptions> = {
-  async $afterSource(pages, { ignorePages, pageExtensions }) {
+  async $afterSource(pages, { ignorePages, pageExtensions, config, namespace }) {
     const isNonHiddenPage = createPageTest(ignorePages, pageExtensions);
     let finalSharedConfig;
 
-    const indexPagesWithSharedConfig = pages.filter(
+    const indexPages = pages.filter(
       page =>
         path.posix.basename(page.fullPath, path.posix.extname(page.fullPath)) === 'index' &&
-        page.sharedConfig !== undefined &&
         isNonHiddenPage(page.fullPath)
     );
 
+    const indexPagesWithSharedConfig = indexPages.filter(page => page.sharedConfig !== undefined);
+
+    if (indexPagesWithSharedConfig.length === 0 && indexPages.length > 0) {
+      const rootPath = indexPages[0].fullPath;
+      const applyNamespaceSharedConfig = {
+        [`${namespace}~~${rootPath}`]: {
+          paths: indexPages.map(indexPage => indexPage.fullPath),
+          rootPath,
+          namespace
+        }
+      };
+      config.setData({ applyNamespaceSharedConfig });
```

Comment:
the config object is persisted across plugin lifecycle events.  So this source is writing to the global config int he hope another source that does have a shared config will apply it.

This is picked up in the `afterUpdate` event